pkg/cmd/get/clusters.go

```diff
@@ -21,14 +67,296 @@ func preClustersE(cmd *cobra.Command, args []string) error {
 }
 
 func list(cmd *cobra.Command, args []string) error {
-	provider := cluster.NewProvider(cluster.ProviderWithDocker())
+	clusters, err := populateClusterList()
+	if err != nil {
+		return err
+	} else {
+		// Convert the list of the clusters to Table of clusters
+		printTable(printers.PrintOptions{}, generateClusterTable(clusters))
+		return nil
+	}
+}
+
+func populateClusterList() ([]Cluster, error) {
+	logger := helpers.CmdLogger
+
+	detectOpt, err := util.DetectKindNodeProvider()
+	if err != nil {
+		//logger.Error(err, "failed to detect the provider.")
+		return nil, err
+	}
+
+	kubeConfig, err := helpers.GetKubeConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: Check if we need it or not like also if the new code handle the kubeconfig path passed as parameter
+	_, err = helpers.GetKubeClient(kubeConfig)
+	if err != nil {
+		//logger.Error(err, "failed to create the kube client.")
+		return nil, err
+	}
+
+	config, err := helpers.LoadKubeConfig()
+	if err != nil {
+		//logger.Error(err, "failed to load the kube config.")
+		return nil, err
+	}
+
+	// Create an empty array of clusters to collect the information
+	clusterList := []Cluster{}
+
+	// List the idp builder clusters according to the provider: podman or docker
+	provider := cluster.NewProvider(cluster.ProviderWithLogger(kind.KindLoggerFromLogr(&logger)), detectOpt)
 	clusters, err := provider.List()
 	if err != nil {
-		return fmt.Errorf("failed to list clusters: %w", err)
+		//logger.Error(err, "failed to list clusters.")
+		return nil, err
+	}
+
+	// Populate a list of Kube client for each cluster/context matching a idpbuilder cluster
+	manager, _ := CreateKubeClientForEachIDPCluster(config, clusters)
```

Comment:
Need to handle error here. If an error is returned and manager is nil, we will panic.

Reply:
Done